server/src/main/java/org/elasticsearch/common/blobstore/RetryingInputStream.java

```diff
@@ -322,7 +322,17 @@ public void close() throws IOException {
     @Override
     public long skip(long n) throws IOException {
         ensureOpen();
-        return currentStream.skip(n);
+        final int initialAttempt = attempt;
+        while (true) {
+            // noinspection TryWithIdenticalCatches
+            try {
+                return currentStream.skip(n);
```

Comment:
You could do something more elaborate on retry like re-open at the new offset, but better to keep it simple for now